src/Memory/ILogStorage.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as CLM from '@conversationlearner/models'
+
+export interface LogQueryResult {
+    logDialogs: CLM.LogDialog[]
+    continuationToken: string | undefined
+}
+
+export interface ILogStorage {
+
+    /** Add a log dialog to storage */
+    Add(appId: string, logDialog: CLM.LogDialog): Promise<CLM.LogDialog>
```

Comment:
>Add [](start = 4, length = 3)

Is this upsert operation? or adding a new dialog?

Reply:
adding a new one